requestor/ext/requestor-gwtjackson/src/main/java/io/reinert/requestor/ext/gwtjackson/JsonObjectSerdesSchema.java

```diff
@@ -0,0 +1,339 @@
+import com.squareup.javapoet.FieldSpec;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.ParameterSpec;
+import com.squareup.javapoet.ParameterizedTypeName;
+import com.squareup.javapoet.TypeSpec;
+import com.squareup.javapoet.TypeVariableName;
+
+import io.reinert.requestor.ext.gwtjackson.codegen.FieldAssembler;
+import io.reinert.requestor.ext.gwtjackson.codegen.InnerTypeAssembler;
+import io.reinert.requestor.ext.gwtjackson.codegen.MethodAssembler;
+import io.reinert.requestor.ext.gwtjackson.codegen.TypeInfo;
+import io.reinert.requestor.ext.gwtjackson.meta.requestor.JsonObjectSerdesMeta;
+import io.reinert.requestor.serialization.DeserializationContext;
+import io.reinert.requestor.serialization.SerializationContext;
+import io.reinert.requestor.serialization.json.JsonRecordReader;
+import io.reinert.requestor.serialization.json.JsonRecordWriter;
+
+class JsonObjectSerdesSchema {
+
+    final TypeInfo typeInfo;
+    final MapperInterface mapperInterface = new MapperInterface();
+    final CollectionWriterInterface collectionWriterInterface = new CollectionWriterInterface();
+    final ArrayListReaderInterface arrayListReaderInterface = new ArrayListReaderInterface();
+    final LinkedListReaderInterface linkedListReaderInterface = new LinkedListReaderInterface();
+    final HashSetReaderInterface hashSetReaderInterface = new HashSetReaderInterface();
+    final LinkedHashSetReaderInterface linkedHashSetReaderInterface = new LinkedHashSetReaderInterface();
+    final TreeSetReaderInterface treeSetReaderInterface = new TreeSetReaderInterface();
+    final MapperField mapperField = new MapperField();
+    final CollectionWriterField collectionWriterField = new CollectionWriterField();
+    final ArrayListReaderField arrayListReaderField = new ArrayListReaderField();
+    final LinkedListReaderField linkedListReaderField = new LinkedListReaderField();
+    final HashSetReaderField hashSetReaderField = new HashSetReaderField();
+    final LinkedHashSetReaderField linkedHashSetReaderField = new LinkedHashSetReaderField();
+    final TreeSetReaderField treeSetReaderField = new TreeSetReaderField();
+    final Constructor constructor = new Constructor();
+    final ReadJsonMethod readJsonMethod = new ReadJsonMethod();
+    final WriteJsonMethod writeJsonMethod = new WriteJsonMethod();
+    final DeserializeMethod deserializeMethod = new DeserializeMethod();
+    final DeserializeCollectionMethod deserializeCollectionMethod;
+    final SerializeMethod serializeMethod;
+    final SerializeCollectionMethod serializeCollectionMethod;
+    final GetLinkedListReaderMethod getLinkedListReaderMethod = new GetLinkedListReaderMethod();
+    final GetHashSetReaderMethod getHashSetReaderMethod = new GetHashSetReaderMethod();
+    final GetLinkedHashSetReaderMethod getLinkedHashSetReaderMethod = new GetLinkedHashSetReaderMethod();
+    final GetTreeSetReaderMethod getTreeSetReaderMethod = new GetTreeSetReaderMethod();
```

Comment:
That's a structure that I developed for requestor code generation. I keep a generating class declarations in one class (schema) and content in another (code), then a I merge both in another class (the assembler). This way I can test both declarations and code independently.